core/client/src/test/java/alluxio/client/netty/NettyRemoteBlockReaderTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.netty;
+
+import alluxio.network.protocol.RPCBlockReadResponse;
+import alluxio.network.protocol.RPCErrorResponse;
+import alluxio.network.protocol.RPCFileWriteResponse;
+import alluxio.network.protocol.RPCMessage;
+import alluxio.network.protocol.RPCResponse;
+import alluxio.network.protocol.databuffer.DataBuffer;
+import alluxio.network.protocol.databuffer.DataByteBuffer;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandlerContext;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.powermock.reflect.Whitebox;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.nio.ByteBuffer;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Tests for the {@link NettyRemoteBlockReader} class.
+ */
+public class NettyRemoteBlockReaderTest {
+
+  private NettyRemoteBlockReader mNettyRemoteBlockReader;
+  private Bootstrap mBootstrap;
+  private ClientHandler mClientHandler;
+
+  private InetSocketAddress mInetSocketAddress = new InetSocketAddress(1234);
+  private long mBlockId = 4242L;
+  private long mOffset = 1;
+  private long mLength = 10;
+  private long mLockId = 6;
+  private long mSessionId = 3421;
+
+  /**
+   * Set up.
+   */
+  @Before
+  public void before() throws InterruptedException {
+    mNettyRemoteBlockReader = new NettyRemoteBlockReader();
+
+    mBootstrap = Mockito.mock(Bootstrap.class);
+    mClientHandler = new ClientHandler();
+
+    Whitebox.setInternalState(mNettyRemoteBlockReader, "mClientBootstrap", mBootstrap);
```

Comment:
No, we've switched to using dependency injection instead. @goodluckbot, could you add a constructor to `NettyRemoteBlockReader` which takes a `Bootstrap` and `ClientHandler` as arguments instead of creating them itself? Then we could use that constructor here without needing to use `Whitebox`.